### ISSUE TITLE
Multiple Qualified strings should have unique names instead of `string`, `string2`, etc

### ISSUE THREAD

**Issue by changusmc** [uber/motif#35]
Version 0.0.11

Code sample
```
import javax.inject.Named

@motif.Scope
interface UserOrPathScope {
    fun userIdOrPath(): UserOrPath

    @motif.Objects
    open class Objects {
        internal fun provideUserId(@Named("userId") userId: String,  @Named("path") path: String) = UserOrPath(userId, path)
    }
}

fun createUserOrPathScope(): UserOrPathScope {
    val dependencies = object: UserOrPathScopeImpl.Dependencies {
        override fun string(): String {
            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        }

        override fun string2(): String {
            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        }

    }
    return UserOrPathScopeImpl(dependencies)
}

data class UserOrPath(val userId: String, val path: String)
```

The following sample generates this Motif code
```
package com.dropbox.base.inject;

import dagger.Provides;
import java.lang.Override;
import java.lang.String;
import javax.inject.Named;
import motif.internal.DaggerScope;

public class UserOrPathScopeImpl implements UserOrPathScope {
  private final Component component;

  public UserOrPathScopeImpl(Dependencies dependencies) {
    this.component = DaggerUserOrPathScopeImpl_Component.builder()
        .dependencies(dependencies)
        .module(new Module())
        .build();
  }

  @Override
  public UserOrPath userIdOrPath() {
    return component.userOrPath();
  }

  public interface Dependencies {
    @Named("userId")
    String string();

    @Named("path")
    String string2();
  }

  @DaggerScope
  @dagger.Component(
      dependencies = Dependencies.class,
      modules = Module.class
  )
  interface Component {
    UserOrPath userOrPath();
  }

  private static class Objects extends UserOrPathScope.Objects {
  }

  @dagger.Module
  class Module {
    private final UserOrPathScope.Objects objects = new Objects();

    @Provides
    UserOrPathScope userOrPathScope() {
      return UserOrPathScopeImpl.this;
    }

    @Provides
    @DaggerScope
    UserOrPath userOrPath(@Named("userId") String string, @Named("path") String string2) {
      return objects.provideUserId$inject_debug(string, string2);
    }
  }
}
```

The portion in the generated code that I think could use improvement is 
```
  public interface Dependencies {
    @Named("userId")
    String string();

    @Named("path")
    String string2();
  }
```

Is it possible to name this after the parameter name (not the qualified name) in the event of collisions? 

e.g. 
```
  public interface Dependencies {
    @Named("userId")
    String string_pathOrId();

    @Named("path")
    String string_userId();
  }
```

**Comment by Leland-Takamine** [comment]
Change released in `0.0.14`.

**Comment by Leland-Takamine** [comment]
BTW - Are you commonly implementing the Dependencies interface manually? If so I'd like to understand that use case better.

**Comment by changusmc** [comment]
> BTW - Are you commonly implementing the Dependencies interface manually? If so I'd like to understand that use case better.

I am doing commonly implementing that way at the moment. I would say it's mainly due to lack of expertise with using Motif. 

- How would you pass dependencies from outside the graph to Motif?

**Comment by changusmc** [comment]
> Change released in `0.0.14`.

What is your release cadence?

**Comment by Leland-Takamine** [comment]
> How would you pass dependencies from outside the graph to Motif?

We call these "dynamic dependencies", and typically you'd pass them into the [child method](https://github.com/uber/motif/wiki#child-methods) on the parent Scope. In your case, you could define a `MainScope` as shown below. The parameters passed into the child method are available in the `UserOrPathScope`:

```java
@Scope
interface MainScope {

    UserOrPathScope createUserOrPath(
            @Named("userId") String userId, 
            @Named("path") String path);

    @motif.Objects
    class Objects { ... }
}
```

I'm not sure how you've set up your Scopes, but it's useful to set them up in a "tree" like this, which allows you to expose dependencies defined at the parent down to child Scopes using the [`@Expose`](https://github.com/uber/motif/wiki#expose) annotation. The canonical app structure looks like this:

![image](https://user-images.githubusercontent.com/847683/52392718-f310da00-2a57-11e9-8a38-acdeb0f920fe.png)

Each of the arrows in the graph above is a [child method](https://github.com/uber/motif/wiki#child-methods). Let me know if this makes sense.

> What is your release cadence?

There's no set release schedule. If there's something on master that isn't in the latest release, let us know and we'll cut a new version.
